README.md

```diff
@@ -1,277 +1,899 @@
+Charlie gives Bob 10 ADA and receives 10 AGIX.
+Charlie gives Alice 5 AGIX and receives 10 ADA.
+Charlie now has his original 10 ADA plus an additional 5 AGIX.
+This all occurs in one transaction where Charlie pays the transaction fee.
+```
 
-#### Updating Prices
-The `Update` redeemer allows users to update their asking prices by modifying the `swapPrice` datum field in the "Swap" UTxO(s):
+On net, Charlie pays the transaction fees and receives 5 AGIX in return, while both Alice and Bob's
+swaps are fulfilled.
 
-![Update Swap](./images/Update.jpg)
+##### The Realistic Example
 
-The `Update` redeemer includes checks to ensure the new datum is properly used:
+``` Txt
+Alice has 10 ADA in her swap address and is willing to swap them for 1 DUST/ADA.
+Bob has 10 DUST in his swap address and is willing to swap them for 0.5 AGIX/DUST.
+Charlie has 10 AGIX in his swap address and is willing to swap them for 1 HOSKY/AGIX.
+Mike has 10 HOSKY in his swap address and is willing to swap them for 1 ADA/HOSKY.
+```
 
-1. No Beacons among Tx inputs.
-2. The staking credential must signal approval (via key or script)
-3. All new outputs to the address must contain the proper datum:
-    - The new `SwapPrice` datum must be > 0.
+In this example, Sarah can profitably arbitrage and fulfill all of these swaps like this:
+
+``` Txt
+Sarah looks up all swap addresses willing to swap DUST/ADA. Sarah finds Alice's address.
+Sarah looks up all swap addresses willing to swap AGIX/DUST. Sarah finds Bob's address.
+Sarah looks up all swap addresses willing to swap HOSKY/AGIX. Sarah finds Charlie's address.
+Sarah looks up all swap addresses willing to swap ADA/HOSKY. Sarah finds Mike's address.
+Sarah gives Mike 10 ADA and receives 10 HOSKY.
+Sarah gives Charlie 10 HOSKY and receives 10 AGIX.
+Sarah gives Bob 5 AGIX and receives 10 DUST.
+Sarah gives Alice 10 DUST and receives 10 ADA.
+Sarah now has her original 10 ADA plus an additional 5 AGIX.
+This all occurs in one transaction where Sarah pays the transaction fee.
+```
 
+On net, Sarah pays the transaction fees and receives 5 AGIX in return, while four swaps are
+fulfilled. As shown in this example, Sarah fulfills *both* the AGIX/DUST swap and the HOSKY/AGIX
+swap by "passing through" those pairs on her way back to ADA. As long as the entry and exit pairs
+(in this case ADA/HOSKY and DUST/ADA) have enough liquidity, arbitragers can spread that liquidity
+into less liquid swap pairs. And since two-way swaps naturally incentivize providing liquidity for
+major trading pairs, this requirement for entry and exit liquidity is naturally incentivized to be
+satisfied.
+
+Since the current design is capable of handling 11-14 swaps in a single transaction, there is ample
+flexibility for finding arbitrage opportunities. For example, maybe the 3rd swap is at a slight loss
+but the 5th swap is such a good arbitrage that it more than makes up for it resulting in a net
+profit for the arbitrage.
```

Comment:
This concept of emergent liquidity is a massive opportunity and is enabled by the ease of composing a chain of dependent swaps.  As you indicate in your example, I think this will result in compression of spreads for actively traded pairs in exchange for execution of less liquid pairs.